### DISCUSSION TITLE
Relay Listener Install

### DISCUSSION THREAD

**Discussion by pyrofection** [josephdadams/TallyArbiter#492] [Q&A]
So I am probably just doing this all wrong but I can not for the life of me figure out how to get the relay listener to work with Tally Arbiter. I am attempting to set Tally Arbiter up on a Raspberry PI 4 B+ 4G and am using one of the recommended 8ch relay boards. 

I have attempted to setup Tally Arbiter from the source code but always get an error but am able to install and run Tally Arbiter with out issue using the npm install method. I could really use some help please.

Also the documentation on how to do the installation is very vague and hard for someone like me who doesn't work with these kinds of software installations much.

**Comment by josephdadams** [comment]
Is the TA server running on the same Pi? What version?

**Reply by pyrofection** [reply]
Sorry it's taken me so long to get back to this. Work and life just got crazy busy.

Short answer is yes, I am attempting to run TA with the Relay Listener on the same Raspberry Pi 4 B+ 4 GB unit. I am running the basic Rasbian Pi 32 bit OS (Debian Bullseye 11 branch)

The issue I keep running into with it is when I attempt to perform the `npm install` command in the project root folder of the source code I continue to receive the errors in the attached pictures.

![IMG_2499](https://user-images.githubusercontent.com/64550993/213293011-78d601c6-c4e3-44c1-b92f-9061ef192a33.jpg)

When installing TA via the CLI instructions I have no problems and am able to run TA but am struggling to install the Relay Listener.

**Reply by hrueger** [reply]
Hi, what version of node and npm are you using? `node -v` / `npm -v`

**Reply by pyrofection** [reply]
node 16.13.1 npm 8.1.2 

Did a forced cache clean on npm and reinstalled the 16.13.1 using n and it allowed me to install TA but did not install the Relay Listener 

I feel like I am missing something on how to get the relay listener to install.

**Reply by josephdadams** [reply]
It is probably a path issue with node and npm. Can you install semver manually?

**Reply by pyrofection** [reply]
Doing a clean install of everything now and manually installing semver and will keep you posted.

**Reply by pyrofection** [reply]
Ok so following the the "Build and run from source" section of the install documentation I am able to get TA installed and working with the manual install of semver but am not seeing the relay listener start up or install. Am I missing a step here?

Install process

1. Fully update OS by running `sudo apt-get update` `sudo apt-get upgrade -y` `sudo apt-get autoclean -y`
2. Install Node.js and NPM via `sudo apt-get install node.js -y` `sudo apt-get install npm -y`
3. Force update/upgrade Node.js and NPM using n via the following commands `sudo npm cache clean -f` `sudo npm install -g n` `sudo n 16.13.1`
4. Install the USB lib using `sudo apt install libudev-dev libusb-1.0-0-dev`
5. Run the git clone command `git clone https://github.com/josephdadams/tallyarbiter`
6. Change to the project root folder via `cd tallyarbiter`
7. Run `sudo npm install`
8. Run `sudo npm start`

Not sure if I need to run anything else to install the Relay Listener

**Reply by KlausKuka** [reply]
Is there such a nice guide for the Windows version? I have the same problem. Unfortunately, I can't do anything with the instructions on GitHub.
